docs/topics/Permissions.md

```diff
@@ -39,18 +41,47 @@ Permissions in Discord are stored within a 53-bit integer and are calculated usi
 
 **\* These permissions require the owner account to use [two-factor authentication](#DOCS_OAUTH2/twofactor-authentication-requirement) when used on a guild that has server-wide 2FA enabled.**
 
+Note that these internal permission names may be referred to differently by the Discord client. For example, "Read Text Channels and See Voice Channels" refers to READ_MESSAGES; "Manage Permissions" refers to MANAGE_ROLES, and "Use Voice Activity" refers to USE_VAD.
+
 ## Permission Hierarchy
 
-Permissions follow a hierarchy with the following roles:
+How permissions apply may at first seem intuitive, but there are some hidden restrictions that prevent bots from performing certain inappropriate actions based on a bot's highest role compared to its target's highest role. A bot's or user's highest role is its role that has the greatest position value in the guild, with the default @everyone role starting at 0. Permissions follow a hierarchy with the following rules:
 
 * A bot can grant roles to other users that are of a lower position than their highest role.
 * A bot can edit roles of a lower position than their highest role, but they can only grant permissions they have to those roles.
 * Bots can only sort roles lower than their highest role.
-* Bots can only kick/ban users of with a lower highest role than themselves.
-
-Otherwise, permissions do not obey role hierarchy. For example, a user has two roles: A and B. A denies the `READ_MESSAGE` permission on a #coolstuff channel. B allows the `READ_MESSAGE` permission on the same #coolstuff channel. The user would ultimately be able to read messages from #coolstuff.
-
-There are cases where permission collisions could occur for a user; that is to say, they may have certain roles or overrides with permissions that contradict each other. With this in mind, permissions are applied to users in the following hierarchy:
+* Bots can only kick/ban users that have lower highest role than themselves.
+
+Otherwise, permissions do not obey the role hierarchy. For example, a user has two roles: A and B. A denies the `READ_MESSAGE` permission on a #coolstuff channel. B allows the `READ_MESSAGE` permission on the same #coolstuff channel. The user would ultimately be able to read messages from #coolstuff, regardless of the role positions.
+
+## Permission Overwrites
```

Comment:
I think rather than re-writing a rather large subset of that permissions table again, it would be better to append the first table with "granularity" flags that demonstrate at what level certain permissions can be set. For example:

| Permission | Value | Description | Scope |
 |------------|-------|-----------| -------- |
 | CREATE\_INSTANT\_INVITE | `0x00000001` | Allows creation of instant invites | C, U |

>info
>The scope of permissions explains at what level these permissions can be set. `G = Guild`, `C = Channel`, `U = User`.

Reply:
Or something like that

Reply:
I thought about extending the original table, too. Either with new columns, or flags like you mention. So we're on the same page there. I was worried about making the table too wide, though. There's a maximum width and with more columns, the description would get squished too much.